internal/pkg/controllers/profile/profile.go

```diff
@@ -108,23 +110,35 @@ func (r *Reconciler) Reconcile(req reconcile.Request) (reconcile.Result, error)
 		return reconcile.Result{}, errors.Wrap(ignoreNotFound(err), errGetProfile)
 	}
 
+	// Pre-check if the node supports seccomp
+	if !seccomp.New().IsSupported(logger) {
+		err := errors.New("profile not added")
+		reason := fmt.Sprintf(
+			"node %q does not support seccomp",
+			os.Getenv(config.NodeNameEnvKey),
+		)
+		logger.Error(err, "reason", reason)
+		r.record.Event(configMap, event.Warning(event.Reason(reason), err))
+		return reconcile.Result{}, errors.Wrap(err, reason)
+	}
```

Comment:
_Following-up on discussion https://github.com/kubernetes-sigs/seccomp-operator/pull/85#pullrequestreview-463875699_

@hasheddan like this? Do you think we should put the operator in some `excluded` state to not spam the events? This way we could stop the operator reconciling if seccomp is disabled on a node…

Reply:
@saschagrunert definitely 👍 If we just don't return the error here then it should stop trying to reconcile. controller-runtime behaves in the following manner:
1. If an `error` is returned, requeue immediately
2. If `reconcile.Result{ Requeue: true }`, requeue immediately
3. If `reconcile.Result{ RequeueAfter: someTime }`, requeue after `someTime` has elapsed
4. If `reconcile.Result{}, nil`, do not requeue (will be requeued if a change to the object is observed, or after the usually very long reconcile timeout configured for the controller manager)

Reply:
Awesome, thank you for the clarification! :+1: